server/player/player.go

```diff
@@ -2930,6 +2930,11 @@ func (p *Player) useContext() *item.UseContext {
 			srcIt, _ := srcInv.Item(src)
 			dstIt, _ := dstInv.Item(dst)
 
+			// If dstIt is enchanted with curse of binding, do not swap.
+			if _, isCursed := dstIt.Enchantment(enchantment.CurseOfBinding{}); isCursed && !p.GameMode().CreativeInventory() {
```

Comment:
This feels out of place. How could this ever happen? I assume any attempt to move the item out will be cancelled at the session level?

Reply:
It does not get cancelled at the session level.

Alternatively, I could go to modify relevant items' [Use() function](https://github.com/df-mc/dragonfly/blob/b27aea2185a0513e13e0f63b6dcadc9dfcdb9a0f/server/item/helmet.go#L16) and check if the carrier (player) is wearing armour enchanted with curse of binding.